src/adapter/src/session/vars.rs

```diff
@@ -281,7 +280,15 @@ static ALLOWED_CLUSTER_REPLICA_SIZES: Lazy<ServerVar<Vec<String>>> = Lazy::new(|
 static WINDOW_FUNCTIONS: ServerVar<bool> = ServerVar {
     name: UncasedStr::new("window_functions"),
     value: &true,
-    description: "Feature flag indicating whether window functions are enabled.",
+    description: "Feature flag indicating whether window functions are enabled (Materialize).",
+};
+
+/// Boolean flag indicating that the remote configuration was synchronized at
+/// least once with the persistent [SessionVars].
+static CONFIG_HAS_SYNCED_ONCE: ServerVar<bool> = ServerVar {
+    name: UncasedStr::new("config_has_synced_once"),
+    value: &false,
+    description: "Boolean flag indicating that the remote configuration was synchronized at least once (Materialize).",
```

Comment:
I don't think we want this becoming part of the stable interface of Materialize. Can we add a new `internal: bool` flag that hides parameters from `SHOW` and `SHOW ALL` unless you're logged in as `mz_system`?

Reply:
Will do.

Reply:
Done.

Reply:
Done.